### PR TITLE
Implement web service + related changes

### PR DESCRIPTION
**Pull request by JCotton1123**
This is an unfinished PR. I've opened it to see if you'd be open to accepting these changes and to discuss them.

The gist of this update is:

* a RESTful API to control the sockets
* a simple web UI to control the sockets
* log refactoring to support capturing logs for the API

Here's a screenshot of the web UI:

screen shot 2018-03-18 at 3 21 36 pm

### CODE REVIEW

**Comment by sandysound** [comment]
Hey @JCotton1123 sure, aways happy to have some more input. I'll have a look through it in properly in the next couple of days.

File: Orvibo.js [hunk 1]
```diff
@@ -2,23 +2,22 @@ const net = require('net');
 const Packet = require('./OrviboPacket');
 const PacketBuilder = require('./PacketBuilder');
 const Utils = require('./Utils');
-const Settings = require('./OrviboSettings');
```
Comment (sandysound) [inline Orvibo.js:5]: Really like the idea of the config.json but we might need to still support the OrviboSettings file for people who might upgrade the library. Or maybe we can mark it as a breaking change. What do you think?

File: Orvibo.js [hunk 2]
```diff
@@ -139,17 +141,23 @@ Orvibo.prototype.startServer = function() {
...
-                plugPacket.processPacket(ORVIBO_KEY);
-            } else {
-                plugPacket.processPacket(socketData.encryptionKey);
+            try {
+                if (plugPacket.packetTypeText() === 'pk') {
+                    plugPacket.processPacket(ORVIBO_KEY);
+                } else {
+                    plugPacket.processPacket(socketData.encryptionKey);
+                }
+            }
```
Comment (sandysound) [inline Orvibo.js:154]: could you just move the catch up to this line so it follows the bracket

File: Orvibo.js [hunk 3]
```diff
@@ -209,11 +218,11 @@ Orvibo.prototype.toggleSocket = function(uid) {
         socket.write(packet);
 
     } else {
-        console.log('Can not find socket ');
+        logger.log('Can not find socket ');
     }
 };
 
-Orvibo.prototype.getConnectedSocket = function() {
+Orvibo.prototype.getConnectedSockets = function() {
```
Comment (sandysound) [inline Orvibo.js:225]: makes more sense ;)

File: OrviboWeb.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+httpServer.get('/sockets/:uid/', (req, res) => {
+  let uid = req.params.uid;
+  let sockets = orviboServer.getConnectedSockets()
+  let socket = sockets.find((s) => { return s.uid == uid });
+  if(socket == undefined) return res.sendStatus(404);
+  res.json(socket);
+});
+
+httpServer.get('/sockets/:uid/:state', (req, res, next) => {
+  let uid = req.params.uid
```
Comment (sandysound) [inline OrviboWeb.js:50]: just missing a ; here

File: index.js [hunk 5]
```diff
@@ -1,2 +1,2 @@
 // Index file for npm
-module.exports = require('./Orvibo');
```
Comment (sandysound) [inline index.js:2]: We might need to think about this as people who are just wanting to use the framework to connect to sockets using their own software will still just want to use Orvibo.js

**Review by sandysound — COMMENTED** [review]
This is all really cool and some great ideas. I was thinking of starting a seperate project to create a frontend and keeping this project purely about controlling the sockets but I guess it could be useful to have it all in one project. What do you think? Should we split this out the web control stuff into a new a new repo that with this project as a dependency? let me know your thoughts?

**Comment by JCotton1123** [comment]
@sandysound I've been back and forth about whether to move my changes into a new project. Initially I was thinking developing the REST API (sort of) would be a convenient way for most users to interface with the "server". But I think I agree that the API/web service should just be moved into its own project. I am going to close this PR and I'll open a new one with some of Orvibo.js specific changes. I'll review your code comments before opening the PR.

**Pull request closed without merging**